W-14229328-azure-events-hub-kt (#2400)
mulesoft/docs-connectors: azure-event-hubs/1.0/modules/ROOT/pages/azure-event-hubs-connector-studio-configuration.adoc

[[create-mule-project]]
== Create a Mule Project

In Studio, create a new Mule project in which to add and configure the connector:

. In Studio, select *File > New > Mule Project*.
. Enter a name for your Mule project and click *Finish*.

[[add-connector-to-project]]
== Add the Connector to Your Mule Project

Add Anypoint Connector for Azure Event Hubs Messaging (Azure Event Hubs Messaging Connector) to your Mule project to automatically populate the XML code with the connector's namespace and schema location and add the required dependencies to the project's `pom.xml` file:

. In the *Mule Palette* view, click *(X) Search in Exchange*.
. In the *Add Dependencies to Project* window, type `azure event hubs messaging` in the search field.
. Click *Azure Event Hubs Connector* in *Available modules*.
. Click *Add*.
. Click *Finish*.

Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.

[[configure-input-source]]
== Configure a Source

A source initiates a flow when a specified condition is met.

> You can configure one of these sources to use with Azure Event Hubs Messaging:

Reviewer: ```suggestion
You can configure one of these sources to use with Azure Event Hubs Messaging Connector:
```

. On the *General* tab, specify the connection information for the connector.
. On the *TLS* tab, optionally specify the TLS information for the connector.
. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[add-connector-operation]]
== Add a Connector Operation to the Flow

When you add a connector operation to your flow, you are specifying an action for that connector to perform.

To add an operation for Azure Event Hubs Messaging Connector, follow these steps:

. In the *Mule Palette* view, select *Azure Event Hubs* and then select the desired operation.
. Drag the operation onto the Studio canvas to the right of the source.

[[configure-global-element]]
== Configure a Global Element for the Connector

When you configure a connector, configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication that the connector requires to access the target Azure Event Hubs system.

> Azure Event Hubs Connector supports *Sas authentication* and *Active directory authentication*.

Reviewer: I would say "SAS authentication and Active Directory authentication" without the bolding, since this is general information and not UI instructions in this paragraph. SAS is all caps, even though it looks like the GUI has it as "SaS."

To configure the global element for Azure Event Hubs Connector, follow these steps:

. Select the operation in the Studio canvas.
. In the configuration screen for the operation, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. On the *General* tab, in *Connection*, select the authentication method to configure:

> * <<config-sas, Sas authentication>>

Reviewer: Can you change "Sas authentication" to "SAS authentication" here? I think it's OK to do this for the section headings if you have the steps reflect the GUI.

> * <<config-active, Active directory authentication>>

Reviewer: Active Directory authentication

[start=4]
. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.

> . Click *Test Connection* to confirm that Mule can connect with the specified server.

Reviewer: Should "Mule" be "Mule runtime"?

. Click *OK*.

[[config-sas]]

> === Configure Sas Authentication

Reviewer: ```suggestion
=== Configure SAS Authentication
```

> Enter the following information on the *General* tab of the *Global Element Properties* screen to configure *Sas Authentication*:

Reviewer: ```suggestion
Enter the following information on the *General* tab of the *Global Element Properties* screen to configure SAS authentication:
```

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required

> | Namespace a| String |  Event Hubs namespace name. |  | x

Reviewer: Event Hubs > Azure Event Hubs?

> | Shared Access Key Name a| String |  Shared access key given by Azure. |  | x

Reviewer: ```suggestion
| Shared Access Key Name a| String |  Shared access key provided by Azure. |  | x
```

> | Shared Access Key a| String |  Shared access key value given by Azure. |  | x

Reviewer: ```suggestion
| Shared Access Key a| String |  Shared access key value provided by Azure. |  | x
```

> | Event Hub Name a| String |  Event Hub name. |  | x

Reviewer: ```suggestion
| Event Hub Name a| String |  Azure Event Hub name. |  | x
```

| Share Connection a| Boolean |  Toggles the builder to use the same connection for producers or consumers. |  false | 
| Checkpoint Store Type a| One of:

* <<AzureBlobStorageSasAuthentication>>

> * <<AzureBlobStorageActiveDirectoryAuthentication>> | Selects Azure Blob Storage Connection Provider. |  |

Reviewer: ```suggestion
* <<AzureBlobStorageActiveDirectoryAuthentication>> | Selects the Azure Blob Storage Connection Provider. |  |
```

Reviewer: We try not to use "Connection Provider" in the non-core connectors because it is a techie term. I just use "connection': Selects the Azure Blob Storage connection."

> | Reconnection a| Reconnection |  Configures a reconnection strategy to use when a connector operation fails to connect to an external server. |  | 

Reviewer: ```suggestion
| Reconnection a| Reconnection |  Configures a reconnection strategy to use when the connector operation fails to connect to an external server. |  | 
```

|===

[[config-active]]
=== Configure Active Directory Authentication

> Enter the following information on the *General* tab of the *Global Element Properties* screen to configure *Active directory authentication*. 

Reviewer: ```suggestion
Enter the following information on the *General* tab of the *Global Element Properties* screen to configure Active Directory authentication. 
```

> Using this authentication enables you to configure *Token Credential Properties* or *Use default credentials*.

Reviewer: ```suggestion
Using this authentication type enables you to configure *Token Credential Properties* or *Use default credentials*.
```

Reviewer: I suggest removing the bold and writing out the words.

enables you to either configure token credential properties or use default credentials.

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Token Credential Properties a| <<TokenCredentialProperties>> |  Acquires a token via service principal authentication. |  | 

> | Use Azure Default Credential a| Boolean |  Use DefaultAzureCredentials mechanism. |  false | 

Reviewer: ```suggestion
| Use Azure Default Credential a| Boolean |  Use the DefaultAzureCredentials mechanism. |  false | 
```

Reviewer: Can you say "Use the default Azure credentials mechanism"?

> | Namespace a| String |  Event Hubs namespace name. |  | x

Reviewer: Azure Event Hubs? I will stop marking this.

> | Authority Host a| String |  Specifies the Azure active directory host. Defaults to https://login.microsoftonline.com/ |  | 

Reviewer: ```suggestion
| Authority Host a| String |  Specifies the Azure Active Directory host. Defaults to https://login.microsoftonline.com/ |  | 
```

| Event Hub Name a| String |  Event Hub name. |  | x
| Share Connection a| Boolean | Toggles the builder to use the same connection for producers or consumers. |  false | 
| Checkpoint Store Type a| One of:

* <<AzureBlobStorageSasAuthentication>>
* <<AzureBlobStorageActiveDirectoryAuthentication>> |  Select the checkpointing connection provider. |  | 
| Reconnection a| Reconnection |  Configures a reconnection strategy to use when a connector operation fails to connect to an external server. |  | 
|===

[[AzureBlobStorageSasAuthentication]]
=== Configure Azure Blob Storage SAS Authentication

> The following table shows the fields to configure *Azure blob storage sas authentication*.

Reviewer: ```suggestion
The following table shows the fields for configuring Azure blob storage SAS authentication.
```

Reviewer: or "the fields used to configure..."

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Container Name a| String | Container name. |  | x
| Account Name a| String | Account name of the Azure account. |  | x
| Account Key a| String | Account key of the Azure account. |  | x
|===

[[AzureBlobStorageActiveDirectoryAuthentication]]
=== Configure Azure Blob Storage Active Directory Authentication

> The following table shows the fields to configure *Azure blob storage active directory authentication*.

Reviewer: fields for configuring...

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Token Credential Properties a| <<TokenCredentialProperties>> | Acquires a token via service principal authentication. |  | 

> | Use Azure Default Credential a| Boolean | Setting this flag to `true`, DefaultAzureCredential Class(Azure.Identity) is created to get a token to authenticate. | false | 

Reviewer: ```suggestion
| Use Azure Default Credential a| Boolean | If you set this flag to `true`, DefaultAzureCredential Class(Azure.Identity) is created to get a token to authenticate. | false | 
```

Reviewer: Can you write out DefaultAzureCrefential Class?

| Container Name a| String | Container name. |  | x
| Account Name a| String | Account name of the Azure account. |  | x

> | Authority Host a| String |Specifies the Azure active directory host. Defaults to https://login.microsoftonline.com/ |  | 

Reviewer: Not parallel:

Azure Active Directory host.

|===

[[TokenCredentialProperties]]
=== Configure Token Credential Properties

> The following table shows the fields to configure *Token credentials properties*.

Reviewer: ```suggestion
The following table shows the fields for configuring *Token credentials properties*.
```

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Tenant Id a| String | Tenant ID. |  | 
| Client Id a| String | Client ID. |  | 
| Client Secret a| String | Client secret. |  | 
|===

=== Configuration Examples

> The following example shows the configuration for *Sas Authentication* with *Azure blob storage sas authentication* in Studio:

Reviewer: Again, I think that this is better if you write it out without bolding.

> image::aeh-sas.png[Configuring azure blob storage sas authentication]

Reviewer: sas > SAS

In the *Configuration XML* editor, the `azure-eventhubs:sas-authentication-connection` and `azure-eventhubs:azure-blob-storage-sas-authentication` configurations look like this:

[source,xml,linenums]
----
	<azure-eventhubs:azure-eventhubs-config name="sas" doc:name="Azure Event Hubs Connector Azure Event Hubs" >
		<azure-eventhubs:sas-authentication-connection namespace="${config.namespace}" sharedAccessKeyName="${sas-config.sharedAccessKeyName}" sharedAccessKey="${sas-config.sharedAccessKey}" eventHubName="${config.eventHuName}" >
			<azure-eventhubs:checkpoint-store-type >
				<azure-eventhubs:azure-blob-storage-sas-authentication containerName="${bs-config.containerName}" accountName="${bs-config.accountName}" accountKey="${bs-config.accountKey}" />
			</azure-eventhubs:checkpoint-store-type>
		</azure-eventhubs:sas-authentication-connection>
	</azure-eventhubs:azure-eventhubs-config>
----

> The following example shows the configuration for *Active directory authentication* with *Azure blob storage sas authentication* and *Token credentials properties* in Studio:

Reviewer: Ditto

image::aeh-active-directory-client.png[Configuring active directory authentication with client credentials]

In the *Configuration XML* editor, the `azure-eventhubs:active-directory-authentication-connection`, `azure-eventhubs:azure-blob-storage-sas-authentication`, and `azure-eventhubs:token-credential-properties` configurations look like this:

[source,xml,linenums]
----
<azure-eventhubs:azure-eventhubs-config name="ActiveDirectory" doc:name="Azure Event Hubs Connector Azure Event Hubs" >
		<azure-eventhubs:active-directory-authentication-connection namespace="${config.namespace}" eventHubName="${config.eventHubName}" >
			<azure-eventhubs:token-credential-properties tenantId="${ad-config.tenantId}" clientId="${ad-config.clientId}" clientSecret="${ad-config.clientSecret}" />
			<azure-eventhubs:checkpoint-store-type >
				<azure-eventhubs:azure-blob-storage-sas-authentication containerName="${bs.config.containerName}" accountName="${bs.config.accountName}" accountKey="${bs.config.containerName}" />
			</azure-eventhubs:checkpoint-store-type>
		</azure-eventhubs:active-directory-authentication-connection>
	</azure-eventhubs:azure-eventhubs-config>
----

> The following example shows the configuration for *Active directory authentication* with *Azure blob storage active directory authentication* and *Use azure default credentials* in Studio:

Reviewer: Ditto

image::aeh-active-directory-default-client.png[Configuring active directory authentication with default client credentials]

> In the *Configuration XML* editor, the `azure-eventhubs:active-directory-authentication-connection`, `azure-eventhubs:azure-blob-storage-sas-authentication` , and `useAzureDefaultCredential` configurations look like this:

Reviewer: Spacing between words looks off.
